lib/schema.js

```diff
@@ -1,214 +1,111 @@
-  },
-};
-
-const getShorthand = (def) => {
-  const keys = Object.keys(def);
-  for (const key of keys) {
-    if (def[key].shorthand) return key;
-  }
-  return '';
-};
-
-const check = (name, def, val) => {
-  if (def.required === false && val === undefined) return [];
-  const { type, key, value, length } = def;
-  if (type === 'array' || type === 'set') {
-    if (checks[type](val, value)) return [];
-    return [`Field "${name}" expected to be ${type} of ${value}`];
-  } else if (type === 'object' || type === 'map') {
-    if (checks[type](val, key, value)) return [];
-    return [`Field "${name}" expected to be ${type} of ${key}: ${value}`];
-  } else if (type === 'enum') {
-    if (def.enum.includes(val)) return [];
-    return [`Field "${name}" value is not of enum: ${def.enum.join(', ')}`];
-  }
-  if (typeof val !== type) {
-    return [`Field "${name}" is not of expected type: ${type}`];
-  }
-  if (length && val) {
-    const len = val.toString().length;
-    const { min, max } = length;
-    if (min && len < min) return [`Field "${name}" value is too short`];
-    if (max && len > max) return [`Field "${name}" exceeds the maximum length`];
-  }
-  return [];
-};
-
-const shorthand = (def) => {
-  if (typeof def === 'string') return def;
-  else if (Array.isArray(def)) return 'tuple';
-  else if (typeof def.array === 'string') return 'array';
-  else if (typeof def.set === 'string') return 'set';
-  else if (typeof def.object === 'object') return 'object';
-  else if (typeof def.map === 'object') return 'map';
-  else if (Array.isArray(def.enum)) return 'enum';
-  return '';
-};
-
-const formatLength = (length) => {
-  if (typeof length === 'number') return { max: length };
-  if (Array.isArray(length)) return { min: length[0], max: length[1] };
-  return length;
-};
-
-const longFormOnlyTypes = new Set(['enum', 'array', 'set', 'map', 'object']);
-const toLongForm = (type, def, key) => {
-  if (typeof def !== 'object' && longFormOnlyTypes.has(type)) {
-    throw new Error(
-      `Short string only definition of "${key}: '${type}'" is not supported`
-    );
-  }
-  if (type === 'enum') {
-    return Object.assign(def, { type, enum: def.enum });
-  } else if (type === 'array' || type === 'set') {
-    const value = def[type];
-    Reflect.deleteProperty(def, type);
-    return Object.assign(def, { type, value });
-  } else if (type === 'map' || type === 'object') {
-    const col = def[type];
-    Reflect.deleteProperty(def, type);
-    const key = Object.keys(col)[0];
-    const value = col[key];
-    return Object.assign(def, { type, key, value });
-  }
-  return { type };
-};
+const RESERVED = ['validate', 'parse', 'serialize', 'format'];
 
 class Schema {
-  constructor(name, raw, namespaces) {
-    const short = shorthand(raw) || raw.type;
-    const defs = short ? { value: raw } : raw;
-    this.name = name;
-    this.kind = short ? 'scalar' : 'struct';
+  constructor(name, raw, namespaces = []) {
+    Schema.KIND = KIND;
```

Comment:
Why is this here and not along with other static properties?

Reply:
Done